src/Database/Database.php

```diff
@@ -2940,6 +2940,10 @@ public function updateDocument(string $collection, string $id, Document $documen
             foreach ($document as $key => $value) {
                 // Skip the nested documents as they will be checked later in recursions.
                 if (\array_key_exists($key, $relationships)) {
+                    // No need to compare nested documents more than max depth.
+                    if (count($this->relationshipWriteStack) >= Database::RELATION_MAX_DEPTH - 1) {
```

Comment:
Let's keep this simple:

```suggestion
                    if (\count($this->relationshipWriteStack) > Database::RELATION_MAX_DEPTH) {
```

Reply:
Ignore this for now, let's add a test for updating a 3rd level document and ensure the changes are made